src/API.CRUD/Extensions/IMvcCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using API.CRUD.ActionFilters;
+using FluentValidation.AspNetCore;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace API.CRUD.Extensions
+{
+    public static class IMvcCoreBuilderExtensions
+    {
+        public static IMvcCoreBuilder AddCustomFluentValidation(this IMvcCoreBuilder mvcCoreBuilder)
+        {
+            if (mvcCoreBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(mvcCoreBuilder));
+            }
+
+            return mvcCoreBuilder.AddFluentValidation(options =>
+                                 {
+                                     options.RegisterValidatorsFromAssemblyContaining<Startup>();
+                                     options.RunDefaultMvcValidationAfterFluentValidationExecutes = false;
+                                 })
+                                 .AddMvcOptions(options => options.Filters.Add(new ValidateModelAttribute()));
+        }
+
+        public static IMvcCoreBuilder AddCustomJsonFormatters(this IMvcCoreBuilder mvcCoreBuilder)
+        {
+            return mvcCoreBuilder.AddJsonFormatters(options =>
+            {
+                options.NullValueHandling = NullValueHandling.Ignore;
+                options.Formatting = Formatting.Indented;
+                options.Converters.Add(new StringEnumConverter());

```

Comment:
Needs discussion. We both have had differing opinions on this in the past. So let's try and find some resources/proof on why one way is better. I think the basic TLDR is 'magic strings' vs predictable int values, e.g readability (strings are human readable) vs predictability. (e.g case sensitivity, cultures, etc). So unless you make assumptions, ints are more predictable IMO.

We should also chat with the frontend guys to see what they prefer.

Reply:
Happy with this. Assumption: frontend guys _always_ do case insensitive string checks.